### PR TITLE
.simplecov: delete private method

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
delete `SimpleCov::add_not_loaded_files` because of private method and repair `.simplecov`

### CODE REVIEW

File: Library/Homebrew/.simplecov [hunk 1]
```diff
@@ -32,8 +37,12 @@ SimpleCov.start do
     SimpleCov.at_exit do
       # Just save result, but don't write formatted output.
       coverage_result = Coverage.result
-      # TODO: this method is private, find a better way.
-      SimpleCov.send(:add_not_loaded_files, coverage_result)
+      coverage_result_dup = coverage_result.dup
```
Comment (reitermarkus) [inline Library/Homebrew/.simplecov:40]: No need to `.dup` if we're reassigning it afterwards again anyways.

File: Library/Homebrew/.simplecov [hunk 2]
```diff
@@ -31,9 +36,11 @@ SimpleCov.start do
 
     SimpleCov.at_exit do
       # Just save result, but don't write formatted output.
-      coverage_result = Coverage.result
-      # TODO: this method is private, find a better way.
-      SimpleCov.send(:add_not_loaded_files, coverage_result)
+      coverage_result = Coverage.result.dup
```
Comment (hyuraku) [inline Library/Homebrew/.simplecov:39]: without `.dup`, an error (`can't modify frozen Hash (FrozenError)`) happens when `coverage_result[absolute_path] ||= SimpleCov::SimulateCoverage.call(absolute_path)`

File: Library/Homebrew/.simplecov [hunk 3]
```diff
@@ -21,6 +21,11 @@ SimpleCov.start do
     # be quiet, the parent process will be in charge of output and checking coverage totals
     SimpleCov.print_error_status = false
   end
+  excludes = ["test", "vendor"]
+  subdirs = Dir.chdir(SimpleCov.root) { Dir.glob("*") }
+               .reject { |d| d.end_with?(".rb") || excludes.include?(d) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/.simplecov:26]:
```suggestion
  subdirs = Dir.chdir(SimpleCov.root) { Pathname.glob("*") }
               .reject { |p| p.extname ==".rb" || excludes.include?(p.to_s) }
```

File: Library/Homebrew/.simplecov [hunk 4]
```diff
@@ -21,6 +21,11 @@ SimpleCov.start do
...
+               .map { |d| "#{d}/**/*.rb" }.join(",")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/.simplecov:27]:
```suggestion
               .map { |p| "#{p}/**/*.rb" }.join(",")
```

**Comment by MikeMcQuaid** [comment]
Thanks again @hyuraku!

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**